flytekit/bin/entrypoint.py

```diff
@@ -356,8 +381,12 @@ def _execute_map_task(
             # Use the resolver to load the actual task object
             _task_def = resolver_obj.load_task(loader_args=resolver_args)
             if not isinstance(_task_def, PythonFunctionTask):
-                raise Exception("Map tasks cannot be run with instance tasks.")
-            map_task = MapPythonTask(_task_def, max_concurrency)
+                raise Exception("Map tasks cannot be run with instance tasks.", _task_def)
+
+            if is_aws_batch_single_job:
```

Comment:
hmmm... seems weird.  why?

Reply:
yes, why can this not be part of the default way of handling things?